<launch>
    <!-- bag的地址与名称 -->
    <!-- <arg name="bag_filename" default="/home/cw/Slam/bag_flies/intel.bag"/> -->
    <arg name="bag_filename" default="/home/cw/Slam/bag_flies/basic_localization_stage_indexed.bag"/>
    <!-- <arg name="bag_filename" default="/home/cw/bag_file/test001.bag"/> -->

    <!-- 使用bag的时间戳 -->
    <param name="use_sim_time" value="true" />
    <param name="scan_topic" value="base_scan"/>
    <param name="cloud_topic" value="base_scan"/>
    <param name="odom_frame" value="odom"/>
    <param name="laser_frame" value="base_laser"/>
    <!-- <param name="scan_topic" value="/robot_6/scan"/>
    <param name="cloud_topic" value="pretreat_cloud"/>
    <param name="laser_frame" value="/robot_6/laser"/>
    <param name="odom_frame" value="/robot_6/odom"/> -->
    <!-- <param name="scan_topic" value="scan"/>
    <param name="cloud_topic" value="scan"/>
    <param name="laser_frame" value="base_laser_link"/>
    <param name="odom_frame" value="odom"/> -->
    <!-- 启动节点 -->
    <node name="mrobot_frame_front_end_node" pkg="mrobot_frame" type="mrobot_frame_front_end_node" output="screen">
        <param name="laser_frame" value="base_laser"/>
        <!-- <rosparam command="load" file="$(find mrobot_frame)/config/mapper_params.yaml" /> -->
    </node>
    <node name="mrobot_frame_mapping_node" pkg="mrobot_frame" type="mrobot_frame_mapping_node" output="screen" />

    <!-- launch rviz -->
    <node name="rviz" pkg="rviz" type="rviz" required="false"
        args="-d $(find mrobot_frame)/rviz/front_end.rviz" />
    <!-- play bagfile -->
    <node name="playbag" pkg="rosbag" type="play" args="--clock $(arg bag_filename)" />
</launch>